src/test/java/com/datadoghq/sketch/ddsketch/SerializerTest.java

```diff
@@ -0,0 +1,93 @@
+                new LinearlyInterpolatedMapping(0.01),
+                new QuadraticallyInterpolatedMapping(0.01),
+                new CubicallyInterpolatedMapping(0.01)
+        )
+                .flatMap(im -> Stream.<Supplier<Store>>of(
+                        SparseStore::new,
+                        () -> new CollapsingLowestDenseStore(1000),
+                        () -> new CollapsingHighestDenseStore(1000),
+                        UnboundedSizeDenseStore::new,
+                        PaginatedStore::new
+                ).map(s -> (Supplier<DDSketch>) () -> new DDSketch(im, s)))
+                .flatMap(supplier -> Stream.of(
+                        NORMAL.of(0, 1),
+                        NORMAL.of(100, 10),
+                        POISSON.of(0.01),
+                        POISSON.of(0.99),
+                        POINT.of(42),
+                        UNIFORM.of(100)
+                ).map(dist -> Arguments.of(supplier, dist)));
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("sketches")
+    public void testProtobufSerialization(Supplier<DDSketch> sketchSupplier, Distribution distribution) throws InvalidProtocolBufferException {
+        DDSketch sketch = load(sketchSupplier, distribution);
+        ByteBuffer buffer = sketch.serialize();
+        assertEquals(DDSketchProtoBinding.toProto(sketch).getSerializedSize(), buffer.remaining());
+
+        assertArrayEquals(DDSketchProtoBinding.toProto(sketch).toByteArray(), buffer.array());
+        com.datadoghq.sketch.ddsketch.proto.DDSketch proto =
+                com.datadoghq.sketch.ddsketch.proto.DDSketch.parseFrom(buffer);
+
+        DDSketch recovered = DDSketchProtoBinding.fromProto(UnboundedSizeDenseStore::new, proto);
+        assertEquals(sketch.getCount(), recovered.getCount(), 1e-7);
+        assertEquals(sketch.getIndexMapping().relativeAccuracy(),
+                recovered.getIndexMapping().relativeAccuracy(), 1e-7);
+        assertEquals(sketch.getIndexMapping().relativeAccuracy(),
+                recovered.getIndexMapping().relativeAccuracy(), 1e-7);
+        if (null != sketch.getPositiveValueStore()) {
+            verifyStore(sketch.getPositiveValueStore(), recovered.getPositiveValueStore());
+        }
+        if (null != sketch.getNegativeValueStore()) {
+            verifyStore(sketch.getNegativeValueStore(), recovered.getNegativeValueStore());
+        }
+    }
+
+    private void verifyStore(Store expected, Store actual) {
+        Iterator<Bin> expectedIt = expected.getAscendingIterator();
+        Iterator<Bin> actualIt = actual.getAscendingIterator();
+        while (expectedIt.hasNext() && actualIt.hasNext()) {
+            Bin x = expectedIt.next();
+            Bin y = actualIt.next();
+            assertEquals(x.getIndex(), y.getIndex(), expected.getClass().getName());
+            assertEquals(x.getCount(), y.getCount(), 1e-7);
+        }
+        assertFalse(expectedIt.hasNext() || actualIt.hasNext());
```

Comment:
Shouldn't it be `&&`? It is supposed to iterate over non-empty buckets. Alternatively, we can check that the remaining ones have counts close to zero.
```suggestion
        assertFalse(expectedIt.hasNext() && actualIt.hasNext());
```